### ISSUE TITLE
Reflect view-level messages in page title element

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#10464]
### Is your proposal related to a problem?

When we use Django’s messages framework to reflect view-level errors / success states, it would be nice for screen reader users if those messages were also reflected in the page’s `<title>`, so they don’t have to navigate the page to understand its state.

### Describe the solution you'd like

For the messages that are already plain-text, this could be as simple as adding their text in the `<title>`. Or as a first step, we could simply add a descriptor of the type of message, for example:

- "New Standard page - Wagtail - Error" when the page editor reloads after validation errors
- "Editing Standard page: About - Wagtail - Success" after successfully saving a draft

### Describe alternatives you've considered

We could also use a live role to communicate the messages as-is but this strikes me as being potentially very verbose (and potentially obtrusive).

**Comment by lb-** [comment]
You probably saw this but page level (view level) messages already have role=status. Which means it will get an implicit aria live value of polite. I'm not sure what gets read on page focus based on this role though, the status was put there to support dynamic view level message injection.

As for title reflecting these kinds of messages - I've not seen that pattern in the wild. Would be good to get some WCAG references or examples of other sites doing this. Mainly as I'm curious about this pattern for my own work projects but would be nice to justify this approach to take. 

----

https://github.com/wagtail/wagtail/blob/aca340f3686cad74a7e91818c0e3b7bdb49de62c/wagtail/admin/templates/wagtailadmin/base.html#L11

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/status_role

We also have the same role set in the login / authentication pages. 

https://github.com/wagtail/wagtail/blob/aca340f3686cad74a7e91818c0e3b7bdb49de62c/wagtail/admin/templates/wagtailadmin/login.html#L10

**Comment by lb-** [comment]
This issue and specifically this linked comment add some useful context. Apparently JAWS will read out some love regions on page load. 

https://github.com/w3c/aria/issues/1360#issuecomment-739319614

This is from 2020 - not sure if NVDA or VoiceOver have now aligned with that approach - worth testing though. 

If this is the case, we may not need to do much, aside from possibly adding a redundant aria-live property of polite (MDN recommends this).

**Comment by thibaudcolas** [comment]
@lb- yes, having `role="status"` on there is excellent for messages added via JavaScript but as far as I’m aware isn’t/shouldn’t be reflected on page load (nor any other live region). I’ve not tested this but can do so if it helps.

For the page title – the only WCAG guideline on it is [SC 2.4.2 Page Titled](https://www.w3.org/TR/2016/NOTE-UNDERSTANDING-WCAG20-20161007/navigation-mechanisms-title.html), which doesn’t say much. The general goals here are to help people with:

- Identifying what information a given page contains / what it’s for
- Distinguishing between pages within a set (for example when you have lots of tabs opened)

The baseline is to have the page’s title and site name in there, lots of sites will then expand that to include the page state (and it should be at the beginning of the title rather than the end like I suggested).

Here are examples:

- UK GDS recommendation to have [pagination state in search results](https://design-system.service.gov.uk/components/pagination/#for-larger-numbers-of-pages)
- UK GDS recommendation to have ["Error:" to convey a form validation fail](https://design-system.service.gov.uk/patterns/validation/#how-to-tell-the-user-about-validation-errors)
- [Google having the search term as page title](https://www.google.com/search?q=example)

---

Back to Wagtail – starting the page title with "Error" or "Success" will help low-vision screen reader users realise what has happened upon submitting a form similarly to what a sighted user could do. And for all users, it’ll help with scenarios where people have multiple tabs opened with similar page names.
